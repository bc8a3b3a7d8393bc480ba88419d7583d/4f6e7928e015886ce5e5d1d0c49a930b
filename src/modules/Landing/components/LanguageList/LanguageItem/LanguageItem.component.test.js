import React from 'react';
import { shallow } from 'enzyme';
import { testSnapshots } from '../../../../../utils/test.util';

import Component from './LanguageItem.component';

describe('Modules/Landing - Language Item', () => {
  const componentProps = {
    selected: false,
    languageCode: 'vi',
    languageName: 'Tiếng Việt',
    countryCode: 'VN',
    countryName: 'Việt Nam',
    onPress: jest.fn(),
  };

  const render = props => shallow(<Component {...props} />);

  testSnapshots(Component, [
    {
      description: 'should render component when the user has not selected yet',
      props: componentProps,
    },
    {
      description:
        'should render component when the user has selected language',
      props: {
        ...componentProps,
        selected: true,
      },
    },
  ]);

  test('should call onPress prop when the user taps on language item', () => {
    const component = render(componentProps);
    component.find('TouchableWithoutFeedback').simulate('press');

    expect(componentProps.onPress).toBeCalledWith({
      languageCode: 'vi',
      languageName: 'Tiếng Việt',
      countryCode: 'VN',
      countryName: 'Việt Nam',
    });
  });

  test('check default props', () => {
    expect(Component.defaultProps).toEqual(
      expect.objectContaining({
        onPress: expect.any(Function),
        selected: false,
        containerStyle: {},
      }),
    );
  });

  test('check return value of default function props', () => {
    const onPressFunctionResult = Component.defaultProps.onPress();
    expect(onPressFunctionResult).toEqual(undefined);
  });
});
