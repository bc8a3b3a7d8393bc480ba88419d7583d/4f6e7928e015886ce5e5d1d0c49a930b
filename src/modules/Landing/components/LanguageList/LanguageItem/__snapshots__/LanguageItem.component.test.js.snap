// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modules/Landing - Language Item snapshots should render component when the user has not selected yet 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "flexDirection": "row",
          "padding": 10,
        },
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.2)",
        },
        Object {},
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/flags/VN.png",
        }
      }
      style={
        Object {
          "height": 20,
          "marginRight": 10,
          "width": 25,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "flex": 1,
            "fontFamily": "Oswald-Regular",
            "fontSize": 12,
          },
          Object {
            "color": "white",
          },
        ]
      }
    >
      Việt Nam (Tiếng Việt)
    </Text>
    <Component />
  </Component>
</TouchableWithoutFeedback>
`;

exports[`Modules/Landing - Language Item snapshots should render component when the user has selected language 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "flexDirection": "row",
          "padding": 10,
        },
        Object {
          "backgroundColor": "white",
        },
        Object {},
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/flags/VN.png",
        }
      }
      style={
        Object {
          "height": 20,
          "marginRight": 10,
          "width": 25,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "flex": 1,
            "fontFamily": "Oswald-Regular",
            "fontSize": 12,
          },
          Object {
            "color": "black",
          },
        ]
      }
    >
      Việt Nam (Tiếng Việt)
    </Text>
    <Component
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2B9FDC",
          "borderRadius": 10,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="white"
        name="check"
        size={12}
      />
    </Component>
  </Component>
</TouchableWithoutFeedback>
`;
